cmake_uninstall.cmake.in

```diff
@@ -1,17 +1,41 @@
-# TODO: doesn't remove directories created for installed files
-if(NOT EXISTS "@CMAKE_BINARY_DIR@/install_manifest.txt")
-  message(FATAL_ERROR "Cannot find install manifest: @CMAKE_BINARY_DIR@/install_manifest.txt")
+if(CMAKE_INSTALL_COMPONENT)
+  set(CMAKE_INSTALL_MANIFEST "install_manifest_${CMAKE_INSTALL_COMPONENT}.txt")
+else()
+  set(CMAKE_INSTALL_MANIFEST "install_manifest.txt")
 endif()
 
-file(READ "@CMAKE_BINARY_DIR@/install_manifest.txt" files)
+if(NOT EXISTS "${CMAKE_BINARY_DIR}/${CMAKE_INSTALL_MANIFEST}")
+  # It's a bit silly to install right before uninstalling, but it appears to be
+  # the only way to generate the manifest file that we need. We are going to
+  # delete the exact same files that we install, so there is no ill behavior
+  # here, just using a few more CPU cycles.
```

Comment:
Yes, and it will even *create* those directories, which is kind of bad.

But this would be better to fix in the uninstall section itself, rather than here, so we can cover both cases.

Reply:
Yeah, I know, I just thought it would make sense to mention it in the comment because having `make uninstall` create directories on the system is what I would call an ill behavior.

Reply:
I added a note about this locally; will include it in my next PR.